feat(bedrock): API pagination

sonofagl1tch opened the pull request:
### Context

The current implementation of the AWS Bedrock service in Prowler only fetches the first page of results for Guardrails and Agents. This limitation means that in environments with a large number of these resources (exceeding the default AWS page size), Prowler fails to detect and scan all of them, leading to incomplete security assessments.

### Description

This PR implements proper pagination logic for the [list_guardrails](cci:1://file:///prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:54:4-84:13) and [list_agents](cci:1://file:///prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:143:4-177:13) methods in [bedrock_service.py](cci:7://file:///prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:0:0-0:0).

Changes include:
- Updated [[prowler/providers/aws/services/bedrock/bedrock_service.py](cci:7://file:///prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:0:0-0:0)](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/services/bedrock/bedrock_service.py) logic to iterate through all available pages of results using the `nextToken` provided by the AWS API.
- Added a new unit test file [[tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py](cci:7://file:///prowler/tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py:0:0-0:0)](https://github.com/prowler-cloud/prowler/blob/master/tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py) to verify that the pagination logic correctly collects resources across multiple pages.
- Added inline documentation to the code to explain the pagination flow and loop mechanics.

### Steps to review

1.  **Code Review**: Verify changes in [[prowler/providers/aws/services/bedrock/bedrock_service.py](cci:7://file:///prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:0:0-0:0)](https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/services/bedrock/bedrock_service.py). Ensure the `while True` loops correctly handle the `nextToken` for both Guardrails and Agents.
2.  **Test Verification**:
    - Run the new unit tests:
      ```bash
      poetry run pytest tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py
      ```
    - Run existing tests to ensure no regressions:
      ```bash
      poetry run pytest tests/providers/aws/services/bedrock/bedrock_service_test.py
      ```

### Checklist

- Are there new checks included in this PR? **No**
    - If so, do we need to update permissions for the provider? N/A
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### UI
N/A

#### API
N/A

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.


closes https://github.com/prowler-cloud/prowler/issues/9607

sonofagl1tch committed:
feat(bedrock): implement pagination for guardrails and agents
- Implemented pagination logic in [_list_guardrails](cci:1://file:///Users/master/git/prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:54:4-84:13) and [_list_agents](cci:1://file:///Users/master/git/prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:143:4-177:13) methods using `nextToken` to ensure all resources are retrieved.
- Added new unit test [tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py](cci:7://file:///Users/master/git/prowler/tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py:0:0-0:0) to verify multi-page resource listing.
- Added inline comments to explain pagination flow.

sonofagl1tch committed:
feat(bedrock): implement pagination for guardrails and agents
- Implemented pagination logic in [_list_guardrails](cci:1://file:///Users/master/git/prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:54:4-84:13) and [_list_agents](cci:1://file:///Users/master/git/prowler/prowler/providers/aws/services/bedrock/bedrock_service.py:143:4-177:13) methods using `nextToken` to ensure all resources are retrieved.
- Added new unit test [tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py](cci:7://file:///Users/master/git/prowler/tests/providers/aws/services/bedrock/bedrock_service_pagination_test.py:0:0-0:0) to verify multi-page resource listing.
- Added inline comments to explain pagination flow.

jfagoagas reviewed (changes requested):
That's fantastic @sonofagl1tch. I left some comments/questions. Don't forget to add a changelog entry and review linters. Let me know if you need help with something else.

sonofagl1tch committed:
Delete api/docs/findings-metadata-optimization-security-review.md

removed from commit as it is out of scope

sonofagl1tch committed:
Delete api/docs/findings-metadata-optimization.md

removing unneeded file from testing

sonofagl1tch committed:
Delete start-prowler.sh

removing unneeded file from testing

sonofagl1tch committed:
Delete stop-prowler.sh

removing unneeded file from testing

sonofagl1tch committed:
refactor(bedrock): Use boto3 paginators for listing guardrails and agents

Replace manual nextToken pagination with boto3's built-in paginators
for _list_guardrails and _list_agents methods. This aligns with the
established pattern used across other Prowler AWS services.

sonofagl1tch committed:
test: merge bedrock service pagination tests into main test file

sonofagl1tch committed:
style: format bedrock service test file with black

sonofagl1tch committed:
style: format bedrock service file with black

sonofagl1tch committed:
test: update Bedrock service test pagination logic

Update changelog to document Bedrock service test improvements

sonofagl1tch committed:
Merge branch 'master' into prowler-aiguardrails

jfagoagas committed:
chore: changelog

jfagoagas reviewed (approved):
Thanks for taking care of this @sonofagl1tch 🏅
